code/modules/mob/living/carbon/human/examine.dm

```diff
@@ -223,14 +223,24 @@
 	var/distance = get_dist(user,src)
 	if(istype(user, /mob/abstract/observer) || user.stat == 2) // ghosts can see anything
 		distance = 1
-	if (src.stat && !(src.species.flags & NO_BLOOD))	// No point checking pulse of a species that doesn't have one.
+	var/mob/living/carbon/human/H = user
+	if(H.has_stethoscope_active() && Adjacent(user))
+		spawn (0)
```

Comment:
Nooooo. Use INVOKE_ASYNC instead.

Reply:
Should I also change the other ``spawn (0)`` under the line ``if(istype(user, /mob/living/carbon/human) && !user.stat && Adjacent(user) && !H.has_stethoscope_active())``?

Reply:
Oh, actually. Would it be better to make checking the pulse/checking with the stet a different proc to ``INVOKE_ASYNC`` by the end so it's cleaner? This way I can also take out the other ``spawn (0)``